win32/src/win32print/win32print.cpp

```diff
@@ -469,6 +484,15 @@ BOOL PyWinObject_AsPRINTER_INFO(DWORD level, PyObject *obinfo, LPBYTE *pbuf)
                   PyWinObject_AsTCHAR(obPortName, &pi5->pPortName, TRUE);
             break;
         }
+        case 6: {
```

Comment:
Thanks, but I don't quite understand why this uses a macro - it would make sense if some of the other cases also used it, but given it's used exactly once it might as well be inline? Ditto for the _BREAK_ON_ERROR - it's always true?

Reply:
Yes, I should probably have written down what was in my head.

Using the macro in this one place only is just the 1<sup>st</sup> step. Once it's accepted (if it will be), I will apply to the other (5) cases as well (and the function code shortened). Anyway, I feel that this should be part of a separate commit.

Regarding *\_BREAK\_ON\_ERROR*: I wanted to be generic, so the macro could also be called from outside a *switch* (or loop). Maybe it's overengineered. I can get rid of it, if requested.

Reply:
I don't think _BREAK_ON_ERROR has value - I can't see how it would ever be called anywhere else. Re the macro itself, I'm really not too bothered by that - a few lines saved can't really hurt, but I also don't see much value given we don't look like adding any new structs any time soon - so I'd be fine with either getting rid of the macro, or moving all others to use it, just not this middle ground :)

Reply:
Removed *\_BREAK\_ON\_ERROR*, and applied the macro in the other places. Also added a small test file.